test/unit/configuration-test.js

```diff
@@ -614,31 +430,49 @@ describe('configuration._coerceRemovedOptions()', () => {
         content: 'FORMAT: 1A\n# Sample API v2\n',
       }],
     };
-    let coerceResult;
-
-    before(() => {
-      coerceResult = configuration._coerceRemovedOptions(config);
-    });
+    const normalizedConfig = configuration._normalizeConfig(config);
+    const { warnings, errors } = configuration._validateConfig(config);
 
     it('gets reformatted', () => {
-      assert.deepEqual(config, {
-        apiDescriptions: [
-          {
-            location: 'configuration.apiDescriptions[0]',
-            content: 'FORMAT: 1A\n# Sample API v2\n',
-          },
-          {
-            location: 'filename.api',
-            content: 'FORMAT: 1A\n# Sample API v1\n',
-          },
-        ],
-      });
+      assert.deepEqual(normalizedConfig.apiDescriptions, [
+        {
+          location: 'configuration.apiDescriptions[0]',
+          content: 'FORMAT: 1A\n# Sample API v2\n',
+        },
+        {
+          location: 'filename.api',
+          content: 'FORMAT: 1A\n# Sample API v1\n',
+        },
+      ]);
+    });
+    it('produces one warning', () => {
+      assert.lengthOf(warnings, 1);
     });
     it('produces no errors', () => {
-      assert.deepEqual(coerceResult.errors, []);
+      assert.lengthOf(errors, 0);
     });
-    it('produces one warning', () => {
-      assert.lengthOf(coerceResult.warnings, 1);
+  });
+});
+
+describe('configuration._resolveConfig()', () => {
+  it('retains class instances', () => {
+    const normalizedConfig = configuration._resolveConfig({});
```

Comment:
Would be nice to also test whether providing a custom emitter works (if I'm reading the test correctly, it only verifies whether the default emitter gets through okay)

Reply:
Agree. There are a few branches of configuration handling logic that were not tested previously.